File: test/Z.Core.Test/System.Random/Random.OneOf.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Z.Core.Test
{
    [TestClass]
    public class System_Random_OneOf
    {
        [TestMethod]
        public void OneOf()
        {
            // Type
            var @this = new Random();

            // Examples
            int value1 = @this.OneOf(1, 2, 3, 4); // return one of this value at random.
            string value2 = @this.OneOf("a", "b", "c", "d"); // return one of this value at random.
            object value3 = @this.OneOf(1, "a", DateTime.Now, new object()); // return one of this value at random.
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.Random/Random.OneOf.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library


Public Module Extensions_465

	''' <summary>
	'''     A Random extension method that return a random value from the specified values.
	''' </summary>
	''' <typeparam name="T">Generic type parameter.</typeparam>
	''' <param name="this">The @this to act on.</param>
	''' <param name="values">A variable-length parameters list containing arguments.</param>
	''' <returns>One of the specified value.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function OneOf(Of T)(this As Random, ParamArray values As T()) As T
		Return values(this.[Next](values.Length))
	End Function
End Module


